tvl2018.py

```diff
@@ -0,0 +1,1269 @@
+    loudness_length = len(np.arange(1.75, 39.25, 0.25))
+
+    instantaneous_specific_loudness_left = np.zeros(
+        (n_segments_in_signal + 1, loudness_length))
+    instantaneous_specific_loudness_right = np.zeros(
+        (n_segments_in_signal + 1, loudness_length))
+
+    for i_segment in range(n_segments_in_signal + 1):
+        """
+        # Display progress every 50 segments
+        if iSegment % 50 == 0:
+            print(f"{iSegment} ms of {nSegmentsInSignal} ms analyzed")
+        """
+
+        segment_start = i_segment * n_samples_per_segment
+        segment_end = segment_start + npts
+        segment = signal[segment_start:segment_end, :]
+
+        # Calculate relevant frequencies and levels
+        f_left_relevant, l_left_relevant, f_right_relevant, l_right_relevant = \
+            signal_segment_to_spectrum(segment, rate, db_max, w_hann, v_limiting_indices)
+
+        if l_left_relevant.size == 0:
+            specific_loudness_left = np.zeros(150)
+        else:
+            excitation_levels_left = \
+                spectrum_to_excitation_pattern_025(f_left_relevant, l_left_relevant)
+            specific_loudness_left = \
+                excitation_to_specific_loudness_binaural_025(excitation_levels_left)
+
+        if l_right_relevant.size == 0:
+            specific_loudness_right = np.zeros(150)
+        else:
+            excitation_levels_right = \
+                spectrum_to_excitation_pattern_025(f_right_relevant, l_right_relevant)
+            specific_loudness_right = \
+                excitation_to_specific_loudness_binaural_025(excitation_levels_right)
+
+        instantaneous_specific_loudness_left[i_segment, :] = specific_loudness_left
+        instantaneous_specific_loudness_right[i_segment, :] = specific_loudness_right
+
+    return instantaneous_specific_loudness_left, instantaneous_specific_loudness_right
+
+
+def shortterm_loudness_to_longterm_loudness(Nst: np.ndarray) -> np.ndarray:
+    """
+    Convert short-term loudness to long-term
+    loudness using AGC with attack and release parameters.
+
+    Args:
+        Nst: Short-term loudness values.
+
+    Returns:
+        Nlt: Long-term loudness values.
+    """
+    aA = 0.01
+    aR = 0.00133
+
+    Nlt = np.zeros(len(Nst))
+
+    Nlt[0] = agc_next_frame(0, Nst[0], aA, aR)
+    for i in range(1, len(Nst)):
+        Nlt[i] = agc_next_frame(Nlt[i-1], Nst[i], aA, aR)
+
+    return Nlt
+
+
+def sound_field_to_cochlea(s: np.ndarray, filename_filter: str) -> np.ndarray:
+    """
+    Applies a filter to the sound field data to simulate cochlear processing.
+
+    Args:
+       s: The input sound field data, a 2D array with two columns for stereo signals.
+       filename_filter: The filename of the filter coefficients (MAT-file).
+
+    Returns:
+       out: The filtered output signal.
+    """
+
+    # Load the filter coefficients from the MAT file
+    filter_data = loadmat(f'{filename_filter}')
```

Comment:
This is icky.  Does it make more sense to read in the file once, convert it to text (or csv) and then use that?  Is the filter data too big to be included in the source?